### ISSUE TITLE
NullPointerException reading file from resources or Jar file

### ISSUE THREAD

**Issue by samyc1201** [gluonhq/substrate#754]
I have the code as below:

```
public static void test() {
	
	System.out.println(readFileAsStringFromJar("test.txt"));
	
}

public static String readFileAsStringFromJar(String path) {
	StringBuilder resultBuilder = new StringBuilder("");
	
	try {
		InputStream in = Thread.currentThread().getContextClassLoader().getResourceAsStream(path);
		BufferedReader reader = new BufferedReader(new InputStreamReader(in));
		String line;
		
		while ((line = reader.readLine()) != null) {
			resultBuilder.append(line);
		}
		
		reader.close();
		in.close();
		
	} catch (Exception e) {
		e.printStackTrace();
	}
	
	return resultBuilder.toString();
}
```
Encounter error below:

```
[Wed Aug 26 18:10:26 SGT 2020][INFO] [SUB] java.lang.NullPointerException
[Wed Aug 26 18:10:26 SGT 2020][INFO] [SUB] 	at java.io.Reader.<init>(Reader.java:167)
[Wed Aug 26 18:10:26 SGT 2020][INFO] [SUB] 	at java.io.InputStreamReader.<init>(InputStreamReader.java:72)
[Wed Aug 26 18:10:26 SGT 2020][INFO] [SUB] 	at hellofx.HelloFX.readFileAsStringFromJar(HelloFX.java:119)
[Wed Aug 26 18:10:26 SGT 2020][INFO] [SUB] 	at hellofx.HelloFX.test(HelloFX.java:109)
[Wed Aug 26 18:10:26 SGT 2020][INFO] [SUB] 	at hellofx.HelloFX$1.handle(HelloFX.java:87)
[Wed Aug 26 18:10:26 SGT 2020][INFO] [SUB] 	at com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:86)
[Wed Aug 26 18:10:26 SGT 2020][INFO] [SUB] 	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:234)
[Wed Aug 26 18:10:26 SGT 2020][INFO] [SUB] 	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
[Wed Aug 26 18:10:26 SGT 2020][INFO] [SUB] 	at com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(CompositeEventDispatcher.java:59)
[Wed Aug 26 18:10:26 SGT 2020][INFO] [SUB] 	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
[Wed Aug 26 18:10:26 SGT 2020][INFO] [SUB] 	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
[Wed Aug 26 18:10:26 SGT 2020][INFO] [SUB] 	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
[Wed Aug 26 18:10:26 SGT 2020][INFO] [SUB] 	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
[Wed Aug 26 18:10:26 SGT 2020][INFO] [SUB] 	at com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
[Wed Aug 26 18:10:26 SGT 2020][INFO] [SUB] 	at com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:54)
[Wed Aug 26 18:10:26 SGT 2020][INFO] [SUB] 	at javafx.event.Event.fireEvent(Event.java:198)
[Wed Aug 26 18:10:26 SGT 2020][INFO] [SUB] 	at javafx.scene.Scene$ClickGenerator.postProcess(Scene.java:3597)
[Wed Aug 26 18:10:26 SGT 2020][INFO] [SUB] 	at javafx.scene.Scene$MouseHandler.process(Scene.java:3899)
[Wed Aug 26 18:10:26 SGT 2020][INFO] [SUB] 	at javafx.scene.Scene.processMouseEvent(Scene.java:1885)
[Wed Aug 26 18:10:26 SGT 2020][INFO] [SUB] 	at javafx.scene.Scene$ScenePeerListener.mouseEvent(Scene.java:2618)
[Wed Aug 26 18:10:26 SGT 2020][INFO] [SUB] 	at com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:409)
[Wed Aug 26 18:10:26 SGT 2020][INFO] [SUB] 	at com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:299)
[Wed Aug 26 18:10:26 SGT 2020][INFO] [SUB] 	at java.security.AccessController.doPrivileged(AccessController.java:101)
[Wed Aug 26 18:10:26 SGT 2020][INFO] [SUB] 	at com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleMouseEvent$2(GlassViewEventHandler.java:447)
[Wed Aug 26 18:10:26 SGT 2020][INFO] [SUB] 	at com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(QuantumToolkit.java:412)
[Wed Aug 26 18:10:26 SGT 2020][INFO] [SUB] 	at com.sun.javafx.tk.quantum.GlassViewEventHandler.handleMouseEvent(GlassViewEventHandler.java:446)
[Wed Aug 26 18:10:26 SGT 2020][INFO] [SUB] 	at com.sun.glass.ui.View.handleMouseEvent(View.java:556)
[Wed Aug 26 18:10:26 SGT 2020][INFO] [SUB] 	at com.sun.glass.ui.View.notifyMouse(View.java:942)
[Wed Aug 26 18:10:26 SGT 2020][INFO] [SUB] 	at com.oracle.svm.jni.JNIJavaCallWrappers.jniInvoke_VARARGS:Lcom_sun_glass_ui_View_2_0002enotifyMouse_00028IIIIIIIZZ_00029V(JNIJavaCallWrappers.java:0)

```

I added :

```
<resourcesList>
    <list>.*\\.txt$</list>
</resourcesList>
```

above not working, so I try this:

```
<resourcesList>
    <list>txt</list>
</resourcesList>
```

Both still producing the same error message. 

I have put `test.txt` in many place for testing:

`src/main/resources/test.txt`

`src/ios/test.txt`

`src/ios/resources/test.txt`

`<one of the jar file>/test.txt`

None of these is working. 

I see some issue from GraalVM page, so I change the resource loading to:

`InputStream in = HelloFX.class.getResourceAsStream(path);`

Still not working. Does Gluon substractVM support resource loading actually?

**Comment by jperedadnr** [comment]
`.txt` is not added by default to the resources list (as explained here: https://docs.gluonhq.com/client/#_resourceslist), so this is required either way:
```
<resourcesList>
    <list>.*\\.txt$</list>
</resourcesList>
```
To read the resource, this is how you can do it, providing that the resource is added to `src/main/resources/test.txt`:

```
InputStream in = MainClass.class.getResourceAsStream("/test.txt");
```

**Comment by samyc1201** [comment]
@jperedadnr 

I found that I have to clean the build `mvn clean client:build` instead of running as `mvn client:build`, then it is working now, probably something is cached.

I tried again for this location `src/main/resources/test.txt`, it is working fine, but it is not working if I want to load the file from Jar file. 

So currently GraalVM doesn't support this? https://www.graalvm.org/reference-manual/native-image/Resources/ 

This is quite important actually, some third party libraries include metadata file inside the Jar file to be read. 

Any suggestion to overcome this? Should I manually extract those file and put in `src/main/resources/`?
